pkg/proxy/endpoints.go

```diff
@@ -38,12 +38,34 @@ var supportedEndpointSliceAddressTypes = sets.New[string](
 	string(discovery.AddressTypeIPv6),
 )
 
+type endpointString struct {
+	IP       string
+	Port     int
+	endpoint string
+}
+
+func NewEndpointString(s string) endpointString {
```

Comment:
Currently, the `NewEndpointString` method is mainly used in `*_test.go` files, and I am not sure if it needs to return an `err`